src/ImageSharp/Image/ImageBase{TColor}.cs

```diff
@@ -162,13 +162,15 @@ public virtual PixelAccessor<TColor> Lock()
         internal void SwapPixelsBuffers(PixelAccessor<TColor> pixelSource)
         {
             Guard.NotNull(pixelSource, nameof(pixelSource));
-            Guard.IsTrue(pixelSource.PooledMemory, nameof(pixelSource.PooledMemory), "pixelSource must be using pooled memory");
 
+            // TODO: Was this check really useful? If yes, we can define a bool PixelAccessor<TColor>.IsBoundToImage to re-introduce it:
```

Comment:
The only place where `PixelAccessor<T>` is constructed from an outsider array is `ArrayExtensions.Lock(this TColor[] pixels, int width, int height)`. That method is currently unused, and I don't think it's useful at all. Shouln't we remove it instead?